src/QMCWaveFunctions/tests/test_TrialWaveFunction.cpp

```diff
@@ -358,4 +359,167 @@ TEST_CASE("TrialWaveFunction_diamondC_1x1x1", "[wavefunction]")
 
 }
 
+void setup_He_wavefunction(Communicate* c, ParticleSet** elec_ptr, ParticleSet** ions_ptr, TrialWaveFunction** psi)
+{
+  ParticleSet* elec = new ParticleSet;
```

Comment:
Can we avoid using pointers in this file?

Reply:
Fixed.    Used std::unique_ptr for WaveFunctionFactory, so at least now it will get deleted when the test ends.